### ISSUE TITLE
"User feed by user id" hangs

### ISSUE THREAD

**Issue by coder39248583** [drawrowfly/tiktok-scraper#544]
**Describe the bug**
The "User feed by user id" method hangs. Nothing seems to happen.

## If fields below will be empty then issue will be Ignored and Closed

**To Reproduce**
- Used OS - **Windows 7 64bit**
- NodeJs Version (node -v) - **v12.16.3**
- Scraper Version (tiktok-scraper --version) - **1.4.23**
- CLI or Module - **Module**
- Used method - **user**
- Used options - **("6747935906352907269", { number: 6, by_user_id: true, sessionList: ['sid_tt=mysessionlist;'], proxy: myProxy});**
- Do you run the scraper through Proxy, VPN, VPS? - **proxy**
- If scraping from a profile. Did you checked if profile is Public? Scraper won't work with Private profiles! **it is public**
- Preferably provide URL to the profile,hashtag,video and etc page:
url: **https://www.tiktok.com/@gordonramsayofficial**
user id: **6747935906352907269**

code snippet:

```javascript
// User feed by user id
// Some TikTok user id's are larger then MAX_SAFE_INTEGER, you need to pass user id as a string
(async () => {
    try {
        const posts = await TikTokScraper.user("6747935906352907269", {
            number: 6,
            by_user_id: true,
            sessionList: ['sid_tt=mySidTT;'],
            proxy: myProxy
        });
        console.log(posts);
    } catch (error) {
        console.log(error);
    }
})();
```

**Screenshots**
If applicable, add screenshots to help explain your problem. - **N/A**

It works if I use the username instead of id and set "by_user_id" to false, so I'm confident my sessionlist and proxy settings are not at fault.

The reason I was hoping to scrape by user ids rather than by usernames is because, as I understand it, it would eliminate the need for tiktok-scraper to fetch the id itself with an extra http request. But maybe I have misread the source? My proxy provider charges me by bandwidth, and I'm looking for ways to slim down. Even a little bit would help.

**Comment by Pezhvak** [comment]
same here

**Comment by asportnoy** [comment]
@muneeb331 There's a fix for that in #536.

**Comment by coder39248583** [comment]
@asportnoy I tried applying that fix and it still hangs.

It should be noted that my problem differs from @muneeb331's in that I am using modules, not the command line, and I am using "user feed by _user id_", not "user feed by _username_" (the latter of which works for me).

**Comment by 6c5s4v4t1a4h7n8d2q** [comment]
@asportnoy I have made the change 
![image](https://user-images.githubusercontent.com/67946624/119625597-f143ce00-be3c-11eb-8bbf-47b3222a1520.png)
but still getting the same issue. Any resolved this?

**Comment by asportnoy** [comment]
@6c5s4v4t1a4h7n8d2q I am able to reproduce this issue. Code to reproduce: (this is `tiktok`'s ID)
```js
// Does work
TikTokScraper.getUserProfileInfo("107955", {by_user_id: true, proxy: "..."})

// Doesn't work
TikTokScraper.user("107955", {by_user_id: true, proxy: "..."});

// Does work (you can also omit the option entirely)
TikTokScraper.user("107955", {by_user_id: false, proxy: "..."});
```
I reported the issue on the Discord.

**Comment by drawrowfly** [comment]
I have checked it and it is only working with sec_uid id for example this will work fine:
.user("MS4wLjABAAAAv7iSuuXDJGDvJkmH_vz1qkDZYo1apxgzaxdBSeIuPiM", {by_user_id: true})

**It is not working with regular uid id because TIKTOK API USER FEED endpoint is returning results only with sec_uid**

Regarding hanging, this is not normal and i will patch it

**Comment by irfangerard44** [comment]
i have same problem

**Comment by paulescobar** [comment]
**_Okay, I'm a newb so forgive me if I missed anything in the following..._**

**Describe the bug**
> In previous versions of TikTok scraper, if you batch downloaded via "User ID"...the downloads would be placed in a folder titled with the current "Username" of that user.
> 
> This was excellent default behavior because...
> 1) Scraping via "User ID" handles the problem of name changes.
> 2) Download folders with "Usernames" as titles make it easy for us to identify content.
> 
> Everything worked beautifully last month, when I last followed this tutorial for batch scraping via "User ID"...
> https://github.com/drawrowfly/tiktok-scraper/blob/master/examples/CLI/BatchDownload.md
> 
> **But today when I tried the exact same method, the downloads are placed in a folders titled with the "User ID". The problem is that these numerical titles make it hard to identify the content.** 
> 
> **The original behavior of TikTok-Scraper was much better...placing new downloads in folders titled with the current "Username".**

To Reproduce
Steps to reproduce the behavior:
> Apply the history tracking options of TikTok-Scraper ("-s" & "--historypath") to avoid duplicate downloads.
> 
> On day one, scrape using the "Username" of a TikTok profile (ex. "addisonre").
> 
> On day two, scrape using that profiles "User ID" (ex. "6703550784929793030")
> 
> You will end up with two different folders. The first folder containing contents of "day one" will have the "Username" as a title (ex. addisonre). The second folder containing contents of "day two" will have the title of the "User ID" (ex. "6703550784929793030").
> 
> Because we applied "history tracking" options, older downloads are in the "Username" folder, while the more recent downloads are in the "User ID" folder. So this shows at least TikTok-Scraper recognizes both inputs as representing the same profile. 

Details
```
Used OS
`Windows 8.1 64bit`
NodeJs Version (node -v)
`v14.16.1`
Scraper Version (tiktok-scraper --version)
`1.4.23`
CLI or Module
`CLI`
Used method - my batch text file lists one ID per row (as per Batch tutorial here)...
```
id:1111
id:2222
etc.
```
Used options - my batch file uses the command...
```
call tiktok-scraper from-file "C:\Download\[TikTok-Scraper] [Batch].txt" 1 -d -a 1 --timeout 8000 --filepath "C:\Download" --session-file "C:\Download\[TikTok-Scraper] [Cookie].txt" -n 0 -s --historypath "C:\Download\[TikTok-Scraper] [History]"
PAUSE
```
Do you run the scraper through Proxy, VPN, VPS?
`No`
If scraping from a profile. Did you checked if profile is Public? Scraper won't work with Private profiles!
`Public`
Preferably provide URL to the profile,hashtag,video and etc page:
```
Username addisonre
User ID 6703550784929793030
```

```

**Comment by drawrowfly** [comment]
Read please my comment above !!!

**Comment by paulescobar** [comment]
> Read please my comment above !!!

Forgive me but I couldn't fully understand that post (I'm a newbie, sorry!)

I'm confused because you told the person that the regular ID wasn't working. But in my case it is scraping via the User ID, it just isn't putting the downloads in the "Username" folder.

I did the addisonre experiment today, and it scraped only new files via "User ID". Maybe it works on some accounts and not others?

But in general, should I take your post to mean the problem can't be fixed (and for newbs like myself, we can only batch scrape in CLI via Username)?

**Comment by drawrowfly** [comment]
> > Read please my comment above !!!
> 
> Forgive me but I couldn't fully understand that post (I'm a newbie, sorry!)
> 
> I'm confused because I think you told the person that scraping via User ID wasn't working. But in my case it is scraping via the User ID, it just isn't putting the downloads in the "Username" folder.
> 
> I did the addisonre (yesterday) & id:6703550784929793030 (today) experiment. It respected the history records (done via Username) and scraped only new files via "User ID". Maybe it works on some accounts and not others?
> 
> But in general, should I take your post to mean the problem can't be fixed (and for newbs like myself, we can only batch scrape in CLI via Username)?

use sec_uid id and it will work

**Comment by drawrowfly** [comment]
hanging is solved in the version @1.4.25
